### PR TITLE
docs(embedmd): initial change [OSD-10209]

### PR DESCRIPTION
**Pull request by georgettica**
this embedmd works and it does pretty well

run 'make generate-markdown' to update all of the files

- updated OcmClient to be public for this to work

### CODE REVIEW

**Comment by georgettica** [comment]
related to OSD-10209

**Comment by georgettica** [comment]
we might do this across the docs?

**Comment by georgettica** [comment]
fixed all of the files

File: Makefile [hunk 1]
```diff
@@ -2,19 +2,22 @@
 bin/cobra:
 	GOBIN=$(PWD)/bin go install -mod=readonly $(shell go list -m -f '{{ .Path}}/cobra@{{ .Version }}' github.com/spf13/cobra)
 
-cadctl/cadctl: cadctl/**/*.go pkg/**/*.go go.mod go.sum
-	GOBIN=$(PWD)/cadctl go install -ldflags="-s -w -extldflags=-zrelro -extldflags=-znow" -buildmode=pie -mod=readonly -trimpath $(PWD)/cadctl
+bin/embedmd:
+	GOBIN=$(PWD)/bin go install -mod=readonly github.com/campoy/embedmd@latest
```
Comment (shibumi) [inline Makefile:6]: Can you pin this to a version? If embedmd ever breaks we are wasting time investigating which version did work. 
Just pin it at the version you are currently using

File: pkg/aws/README.md [hunk 2]
```diff
@@ -36,18 +36,20 @@ osdctl account cli -oenv -C ${CLUSTER_INTERNAL_ID} > file.env
...
-AWS_SESSION_TOKEN, hasAWS_SESSION_TOKEN := os.LookupEnv("AWS_SESSION_TOKEN")
-AWS_DEFAULT_REGION, hasAWS_DEFAULT_REGION := os.LookupEnv("AWS_DEFAULT_REGION")
-if !hasAWS_ACCESS_KEY_ID || !hasAWS_SECRET_ACCESS_KEY || !hasAWS_SESSION_TOKEN || !hasAWS_DEFAULT_REGION {
-    return fmt.Errorf("one of the required envvars in the list '(AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_DEFAULT_REGION)' is missing")
-}
 
-a, err := aws.NewClient(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN, AWS_DEFAULT_REGION)
-if err != nil {
-    return fmt.Errorf("could not start awsClient: %w", err)
+[embedmd]:# (../../cadctl/cmd/cluster-missing/cluster-missing.go /\/\/ GetAWSClient/ /^}$/)
```
Comment (shibumi) [inline pkg/aws/README.md:40]: The escaping looks a little bit weird :D but okay

File: pkg/ocm/ocm.go [hunk 3]
```diff
@@ -21,15 +21,15 @@ var (
 	sl_clusterHasGoneMissing = "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/cluster_has_gone_missing.json"
 )
 
-type ocmClient struct {
+type OcmClient struct {
 	conn *sdk.Connection
```
Comment (shibumi) [inline pkg/ocm/ocm.go:25]: If you export OcmClient would it make sense to export Conn and Cfg as well?

File: Makefile [hunk 4]
```diff
@@ -39,9 +50,14 @@ cadctl-install-local: cadctl/cadctl
 
 .PHONY: cadctl-install-local-force
 cadctl-install-local-force: 
-	rm cadctl/cadctl >/dev/null 2>&1  || true
+	rm cadctl/cadctl >/dev/null 2>&1 || true
 	make cadctl-install-local
 
+MARKDOWN_SOURCES := $(shell find $(SOURCEDIR) -name '*.md')
+.PHONY: generate-markdown
+generate-markdown: $(MARKDOWN_SOURCES) bin/embedmd
```
[the remark this answers is not in the document]
Comment (georgettica) [reply Makefile:58]: yes :)

File: pkg/aws/README.md [hunk 5]
```diff
@@ -36,18 +36,20 @@ osdctl account cli -oenv -C ${CLUSTER_INTERNAL_ID} > file.env
...
+```go
+// GetAWSClient will retrieve the AwsClient from the 'aws' package
+func GetAWSClient() (aws.AwsClient, error) {
```
[the remark this answers is not in the document]
Comment (shibumi) [reply pkg/aws/README.md:43]: ah okay. Makes sense :+1:

File: pkg/ocm/ocm.go (hunk 3 above)
Comment (georgettica) [reply pkg/ocm/ocm.go:27]: Added a comment, feel free to TAL

**Comment by shibumi** [comment]
@rafael-azevedo  I have just merged this, because it was blocking other PRs.

**Pull request merged**